src/free.rs

```diff
@@ -128,17 +128,24 @@ pub fn zip<I, J>(i: I, j: J) -> Zip<I::IntoIter, J::IntoIter>
     i.into_iter().zip(j)
 }
 
-/// Create an iterator that first iterates `i` and then `j`.
+/// Returns an iterator combining two iterators consecutively. 
 ///
 /// [`IntoIterator`] enabled version of [`Iterator::chain`].
 ///
+/// ## Example
 /// ```
 /// use itertools::chain;
+/// 
+/// let a = [1, 2, 3];
+/// let b = [4];
+/// let mut result:Vec<i32> = Vec::new();
 ///
-/// for elt in chain(&[1, 2, 3], &[4]) {
-///     /* loop body */
+/// for element in chain(&a , &b) {
+///     result.push(*element);
```

Comment:
Please inline the arguments `a` and `b` into the call to `chain` (i.e. write `chain(&[1,2,3], &[4])`), so that it's visually similar to the resulting `vec![1,2,3,4]`.